src/main/scala/model/world/Reproduction.scala

```diff
@@ -21,23 +22,75 @@ object Reproduction {
+                                                                            JustMutatedAllele(Alleles.LONG_TEETH),
+                                                                            JustMutatedAllele(Alleles.LONG_TEETH))
+      case Mutations.TEETH_MUTATION_RECESSIVE => genesWithMutations :+ Gene( Mutations.TEETH_MUTATION_RECESSIVE.geneKind,
+                                                                             JustMutatedAllele(Alleles.LONG_TEETH),
+                                                                             StandardAllele(Alleles.SHORT_TEETH))
+      case Mutations.EARS_MUTATION_DOMINANT => genesWithMutations :+ Gene( Mutations.EARS_MUTATION_DOMINANT.geneKind,
+                                                                           JustMutatedAllele(Alleles.HIGH_EARS),
+                                                                           JustMutatedAllele(Alleles.HIGH_EARS))
+      case Mutations.EARS_MUTATION_RECESSIVE => genesWithMutations :+ Gene( Mutations.EARS_MUTATION_RECESSIVE.geneKind,
+                                                                            JustMutatedAllele(Alleles.HIGH_EARS),
+                                                                            StandardAllele(Alleles.LOW_EARS))
+      case Mutations.JUMP_MUTATION_DOMINANT => genesWithMutations :+ Gene( Mutations.JUMP_MUTATION_DOMINANT.geneKind,
+                                                                           JustMutatedAllele(Alleles.HIGH_JUMP),
+                                                                           JustMutatedAllele(Alleles.HIGH_JUMP))
+      case Mutations.JUMP_MUTATION_RECESSIVE => genesWithMutations :+ Gene( Mutations.JUMP_MUTATION_RECESSIVE.geneKind,
+                                                                            JustMutatedAllele(Alleles.HIGH_JUMP),
+                                                                            StandardAllele(Alleles.LOW_JUMP))
+    })
+
+    genesWithMutations
+  }
+
   /**
    * @param mom a bunny
    * @param dad another bunny
    * @return the 4 children of the couple, one for each cell of the Punnett's square
    */
-  def generateChildren(mom: Bunny, dad: Bunny): Population = {
+  def generateChildren(mom: Bunny, dad: Bunny)(mutations: Option[List[Mutation]]): Population = {
     var childrenGenotypes = List.fill(CHILDREN_EACH_COUPLE)(PartialGenotype(Map()))
+    var genesWithMutations: List[Gene] = List()
+    if( mutations.isDefined ) {
+       genesWithMutations = introducingMutation(mutations.get)
+    }
+
     Genes.values.foreach(gk => {
       val grandmaMomAllele = mom.genotype(gk).momAllele
       val grandpaMomAllele = mom.genotype(gk).dadAllele
       val grandmaDadAllele = dad.genotype(gk).momAllele
       val grandpaDadAllele = dad.genotype(gk).dadAllele
-      val anotherGene = Random.shuffle(
+      val genesOfReproduction : List[Gene]= List.concat(
```

Comment:
perchè fai concat?

Reply:
Da quel che capisco io, in `introducingMutations` si creano i Geni che hanno subito mutazioni, e qui l'idea era di concatenarli con quelli già presenti che invece non ne hanno subite (?).
Secondo me il fatto di avere una lista genesWithMutations separata fa un po' di confusione, io userei questo foreach su Gene per scorrerli tutti, per ognuno controllerei se è presente la mutazione per tale Gene o meno (attraverso la lista di mutations passata) e creerei 1 coniglietto mutato su 4 in tale caso (usando il JustMutatedAllele). Se non è presenta mutazione, basta lasciare la generazione dei 4 geni così com'è.

Reply:
Esattamente, l'avevo intesa come dice Asia, nel senso che facendo con _Random.shuffle_ la selezione di un Gene tra i proposti (correggetemi se sbaglio) la prima cosa che mi era venuta in mente era quella di aggiungere alla lista dei Geni tra cui scegliere anche quelli con le mutazioni. 
Ma l'idea, a cui non avevo pensato, di Asia mi piace molto di più. Forse in questo modo, quando verranno introdotte dall'utente le mutazioni, se vengono segnalate dal campo del Gene _isMutated_ con True potremo anche non avere più bisogno di passare la lista dei Geni mutati. Faccio alcune prove e ci ragiono su!